app/ui-react/syndesis/src/modules/integrations/components/editor/choice/SelectChoiceConfigModePage.tsx

```diff
@@ -0,0 +1,142 @@
+/*
+ * Copyright (C) 2016 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { getSteps } from '@syndesis/api';
+import * as H from '@syndesis/history';
+import {
+  ButtonLink,
+  IntegrationEditorActionsListItem,
+  IntegrationEditorChooseAction,
+  IntegrationEditorLayout,
+} from '@syndesis/ui';
+import { WithRouteData } from '@syndesis/utils';
+import * as React from 'react';
+import { PageTitle } from '../../../../../shared';
+import { IEditorSidebarProps } from '../EditorSidebar';
+import {
+  IChoiceStepRouteParams,
+  IChoiceStepRouteState,
+  IPageWithEditorBreadcrumb,
+  ISelectConfigModeRouteParams,
+  ISelectConfigModeRouteState,
+} from '../interfaces';
+import { toUIStep, toUIStepCollection } from '../utils';
+
+export interface ISelectChoiceConfigModePageProps extends IPageWithEditorBreadcrumb {
+  cancelHref: (
+    p: ISelectConfigModeRouteParams,
+    s: ISelectConfigModeRouteState
+  ) => H.LocationDescriptor;
+  sidebar: (props: IEditorSidebarProps) => React.ReactNode;
+  selectHref: (
+    p: IChoiceStepRouteParams,
+    s: IChoiceStepRouteState
+  ) => H.LocationDescriptor;
+}
+
+/**
+ * This page shows the list of actions of a connection containing either a
+ * **to** or **from pattern, depending on the specified [position]{@link ISelectConfigModeRouteParams#position}.
+ *
+ * This component expects some [params]{@link ISelectConfigModeRouteParams} and
+ * [state]{@link ISelectConfigModeRouteState} to be properly set in the route
+ * object.
+ *
+ * **Warning:** this component will throw an exception if the route state is
+ * undefined.
+ */
+export class SelectChoiceConfigModePage extends React.Component<ISelectChoiceConfigModePageProps> {
+  public render() {
+    return (
+      <WithRouteData<ISelectConfigModeRouteParams, ISelectConfigModeRouteState>>
```

Comment:
For the future, there is an equivalent hook for this called `useRouteData`. No need to change this now, eventually we'll update the whole routing thing when `react-router` will hit v6.